source/utilities.tex

```diff
@@ -6463,6 +6464,2036 @@
+
+\pnum
+This subclause describes the class template \tcode{unexpected}
+that represents unexpected objects stored in \tcode{expected} objects.
+
+\rSec3[expected.un.object]{Class template \tcode{unexpected}}
+
+\rSec4[expected.un.object.general]{General}
+
+\indexlibraryglobal{unexpected}%
+\begin{codeblock}
+namespace std {
+  template<class E>
+  class unexpected {
+  public:
+    constexpr unexpected(const unexpected&) = default;
+    constexpr unexpected(unexpected&&) = default;
+    template<class... Args>
+      constexpr explicit unexpected(in_place_t, Args&&...);
+    template<class U, class... Args>
+      constexpr explicit unexpected(in_place_t, initializer_list<U>, Args&&...);
+    template<class Err = E>
+      constexpr explicit unexpected(Err&&);
+
+    constexpr unexpected& operator=(const unexpected&) = default;
+    constexpr unexpected& operator=(unexpected&&) = default;
+
+    constexpr const E& value() const& noexcept;
+    constexpr E& value() & noexcept;
+    constexpr const E&& value() const&& noexcept;
+    constexpr E&& value() && noexcept;
+
+    constexpr void swap(unexpected& other) noexcept(@\seebelow@);
+
+    template<class E2>
+      friend constexpr bool operator==(const unexpected&, const unexpected<E2>&);
+
+    friend constexpr void swap(unexpected& x, unexpected& y) noexcept(noexcept(x.swap(y)));
+
+  private:
+    E @\exposid{val}@;               // \expos
+  };
+
+  template<class E> unexpected(E) -> unexpected<E>;
+}
+\end{codeblock}
+
+\pnum
+A program that instantiates the definition of \tcode{unexpected} for
+a non-object type,
+an array type,
+a specialization of \tcode{unexpected}, or
+a cv-qualified type
+is ill-formed.
+
+\rSec4[expected.un.ctor]{Constructors}
+
+\indexlibraryctor{unexpected}%
+\begin{itemdecl}
+template<class Err = E>
+  constexpr explicit unexpected(Err&& e);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{is_same_v<remove_cvref_t<Err>, unexpected>} is \tcode{false}; and
+\item
+\tcode{is_same_v<remove_cvref_t<Err>, in_place_t>} is \tcode{false}; and
+\item
+\tcode{is_constructible_v<E, Err>} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\effects
+Direct-non-list-initializes \exposid{val} with \tcode{std::forward<Err>(e)}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of \exposid{val}.
+\end{itemdescr}
+
+\indexlibraryctor{unexpected}%
+\begin{itemdecl}
+template<class... Args>
+  constexpr explicit unexpected(in_place_t, Args&&...);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\tcode{is_constructible_v<E, Args...>} is \tcode{true}.
+
+\pnum
+\effects
+Direct-non-list-initializes
+\exposid{val} with \tcode{std::forward<Args>(args)...}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of \exposid{val}.
+\end{itemdescr}
+
+\indexlibraryctor{unexpected}%
+\begin{itemdecl}
+template<class U, class... Args>
+  constexpr explicit unexpected(in_place_t, initializer_list<U>, Args&&...);
```

Comment:
```suggestion
  constexpr explicit unexpected(in_place_t, initializer_list<U> il, Args&&... args);
```

Reply:
Nearly all ctors like this need to have named arguments, I think.